concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -1480,245 +1491,522 @@ redelegatePassive accounts (DelegatorId accId) =
-                                bakerInfoEx =
-                                    BaseAccounts.BakerInfoExV1
-                                        { _bieBakerPoolInfo = poolInfo,
-                                          _bieBakerInfo = bakerInfo
-                                        }
-                                updAcc = addAccountBakerV1 bakerInfoEx bcaCapital bcaRestakeEarnings
-                            -- This cannot fail to update the account, since we already looked up the account.
-                            newAccounts <- Accounts.updateAccountsAtIndex' updAcc ai (bspAccounts bsp)
-                            (BCSuccess [] bid,)
-                                <$> storePBS
-                                    pbs
-                                    bsp
-                                        { bspBirkParameters = newBirkParams,
-                                          bspAccounts = newAccounts
-                                        }
-doConfigureBaker pbs ai BakerConfigureUpdate{..} = do
-    origBSP <- loadPBS pbs
-    cp <- lookupCurrentParameters (bspUpdates origBSP)
-    res <- MTL.runExceptT $ MTL.runWriterT $ flip MTL.execStateT origBSP $ do
-        baker <- getAccountOrFail
-        -- Check the various updates are OK, getting the transformation on the account
-        -- implied by each.
-        uKeys <- updateKeys baker
-        uRestake <- updateRestakeEarnings baker
-        uPoolInfo <- updateBakerPoolInfo baker cp
-        uCapital <- updateCapital baker cp
-        -- Compose together the transformations and apply them to the account.
-        let updAcc = uKeys >=> uRestake >=> uPoolInfo >=> uCapital
-        modifyAccount' updAcc
-    case res of
-        Left errorRes -> return (errorRes, pbs)
-        Right (newBSP, changes) -> (BCSuccess changes bid,) <$> storePBS pbs newBSP
+    addValidatorChecks bsp va
+    newBirkParams <- do
+        pab <- refLoad (bspBirkParameters bsp ^. birkActiveBakers)
+        newAggregationKeys <- Trie.insert (bkuAggregationKey vaKeys) () (pab ^. aggregationKeys)
+        newActiveBakers <- Trie.insert bid emptyPersistentActiveDelegators (pab ^. activeBakers)
+        newPABref <-
+            refMake $
+                pab
+                    & aggregationKeys .~ newAggregationKeys
+                    & activeBakers .~ newActiveBakers
+                    & totalActiveCapital %~ addActiveCapital vaCapital
+        return $ bspBirkParameters bsp & birkActiveBakers .~ newPABref
+    let poolInfo =
+            BaseAccounts.BakerPoolInfo
+                { _poolOpenStatus = vaOpenForDelegation,
+                  _poolMetadataUrl = vaMetadataURL,
+                  _poolCommissionRates = vaCommissionRates
+                }
+    let bakerInfo = bakerKeyUpdateToInfo bid vaKeys
+    let bakerInfoEx = BaseAccounts.BakerInfoExV1 bakerInfo poolInfo
+    acc <- fromJust <$> Accounts.indexedAccount ai (bspAccounts bsp)
```

Comment:
I believe it shouldn't happen due to the first precondition

Reply:
But where's the first precondition enforced? I think it would be nice in these cases to point to the code to 'proof' that the error case can't happen.

Reply:
Maybe worth pointing this out in a comment. I think it's generally good practice to always add a comment on the safety when using non-total functions.

Reply:
Added comment that it is safe due to precondtion.